chapters/api_concepts.txt

```diff
@@ -456,6 +456,42 @@ example are the vertex attributes of the
 start with `vertex.` followed by the attribute name.
 ====
 
+Arrays define a valid region of elements, which by default is always the full
+capacity of the array. Implementations can allow applications to keep a singular
+allocation of data, but use a parameter to tell the application only to use a
+subset of elements. This gives applications the ability to more efficiently
+change the number of elements used by parent objects at the cost of memory size
+efficiency.
+
+Implementations provide this functionality by implementing
+<<core_extensions_array1d_region, extension `KHR_ARRAY1D_REGION`>>, which adds
+the following parameter to `ANARIArray1D`:
+
+.Additional parameters accepted by `ANARIArray1D` with <<core_extensions_array1d_region, extension `KHR_ARRAY1D_REGION`>>.
+[cols="<,<,>,<4",options="header,unbreakable"]
+|================================================================
+| Name     | Type         | Description
+| region   |`UINT64_BOX1` | region of elements currently in use
+|================================================================
+
+The element from `region.lower` is _inclusive_, while `region.upper` is
+_exclusive_. This is intended to mimic C++ begin/end iterators on containers.
+
+If `region` is not set, all elements are used as specified when the array was
+constructed. When `region` is set, it is clamped to the range of the array's
+constructed size respectively and warnings should be emitted by the debug layer
+if `region.upper` is larger than capacity.
+
+It is undefined behavior for `region.lower` >= `region.upper`, meaning
+implementations should be able to rely on 1) the array containing at least one
+element, and 2) `region.lower` < `region.upper`.
+
+Mapping an array always returns the full capacity, regardless of the elements
+specified by `region`. Object handles within `region` must always be valid, but
+handles outside of `region` may be invalid. Object arrays can be constructed
```

Comment:
This contradicts (or relaxes?) the spec further above
> ANARIArray objects containing object handles increase the ref count of all objects in the array.
Reference counts are updated on creation of the array object and when unmapping the array.

Reply:
This is orthogonal to object lifetime concerns: valid handles should always have their lifetime managed correctly. This statement in the extension is just guaranteeing that there aren't invalid handles within the range actually in use (i.e. conditional iteration wherever the data is used by the implementation).

Basically this clause lets you create a managed array of objects and only map/write/unmap values into `region` instead of the entire array.

Reply:
OK, I thought that "ANARIArray objects containing object handles increase the ref count of all objects in the array. Reference counts are updated on creation of the array object and when unmapping the array" _implies_ that _all_ handles need to be valid (on creation and on unmap). Which cannot be true for managed arrays on creation... So, it actually means that `NULL` (invalid) handles are ignored during lifetime management. But using invalid `NULL` handles during rendering would be an error. The region extension allows to reduce the active array size, the used elements. So, strictly speaking, this clarification (that handles outside the region can be invalid) is redundant?

Reply:
Yes, I removed the redundant part in 34a4e80.